src/routes/session.ts

```diff
@@ -1,29 +1,79 @@
+import * as _ from 'lodash';
 import {
 	loginUserXHR,
 	findUser,
 	getUser,
 	comparePassword,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { User as DbUser } from '../models';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	try {
+		const user = await getUser(req);
+
+		// If the user email is available, then the record was successfully
+		// populated by the passport middleware
+		if (user.email !== undefined) {
```

Comment:
`user` can be undefined here, in which case this would throw an error of trying to access a property on undefined.  I get that the try/catch will technically handle that but I feel like that's by accident rather than design - if you want to ensure it's not undefined then you can use
```ts
const user = await getUser(req, true);